model/src/test/kotlin/licenses/ResolvedLicenseInfoTest.kt

```diff
@@ -143,6 +149,177 @@ class ResolvedLicenseInfoTest : WordSpec() {
                 filteredResolvedLicenseInfo.licenses should containLicensesExactly(mit, bsd)
             }
         }
+
+        "filterLicenseResolutions()" should {
+            "remove license that is resolved by a license rule violation resolution" {
+                val resolvedLicenseInfo = ResolvedLicenseInfo(
+                    id = Identifier.EMPTY,
+                    licenseInfo = mockk(),
+                    licenses = listOf(
+                        ResolvedLicense(
+                            license = gpl.toSpdx() as SpdxSingleLicenseExpression,
+                            originalDeclaredLicenses = setOf(gpl),
+                            originalExpressions = mapOf(
+                                LicenseSource.DECLARED to setOf(gpl.toSpdx()),
+                                LicenseSource.DETECTED to setOf(gpl.toSpdx())
+                            ),
+                            locations = emptySet()
+                        )
+                    ),
+                    copyrightGarbage = emptyMap(),
+                    unmatchedCopyrights = emptyMap()
+                )
+
+                val violations = listOf(
+                    RuleViolation(
+                        "example rule",
+                        Identifier("Maven", "org.oss-review-toolkit", "example", "0.0.1"),
+                        SpdxSingleLicenseExpression.parse(gpl),
+                        null,
+                        Severity.ERROR,
+                        "example message",
+                        "how to fix"
+                    )
+                )
+
+                val resolutions = listOf(
+                    RuleViolationResolution(
+                        "example message", RuleViolationResolutionReason.CANT_FIX_EXCEPTION, "example comment"
+                    ),
+                    RuleViolationResolution(
```

Comment:
I think, this second resolution does not bring additional value to the test coverage, as there is a dedicated test case later to check whether only matching resolutions are taken into account.